### DISCUSSION TITLE
Using Labs to Stop Video Recordings that were Not Started by Labs

### DISCUSSION THREAD

**Discussion by AdamVen** [gopro/labs#828] [Q&A] [answered]
The short version of the question is: 
Is it possible to stop video recording using Labs when the recording itself was not initialized by Labs? 

Long version:
I have my GoPro attached via USB to one of the ports on a Raspberry Pi onboard a robotic platform, and I'm using the Labs command MQRDR=1MWAKE=2!MTUSB=1!MBOOT="!Lboottest"!uN!uE!R to power on my Hero10 Black when the Pi is powered on. I'm then using a script on the Raspberry Pi to initialize a connection using Open GoPro, to get a copy of the file list and start recording a new video using the wired connection. This all works great, but I can't seem to get the GoPro to stop recording upon USB power disconnect. I was hoping that including the USB Power Trigger commands at the end would stop the recording when the USB connection drops. 

However, as far as I can tell, because the recording wasn't initialized using a !S command between the !uN and !uE commands, the !uE command doesn't know that it needs to stop recording. I've attempted to use the command !uN!6000N!uE!R to set a simple pause between the USB power trigger commands hoping that just having something between the USB Power Trigger commands would help, but this appears to lock up the GoPro in a 'busy' mode and doesn't allow connection via Open GoPro. 

I think this belongs in the Labs section because I'm assuming that starting recording via Open GoPro is functionally equivalent to starting recording using the button on the camera. 

More Details/Context:
Using Open GoPro via USB to get the file list and initialize recording is a hard requirement, as I need a Pi system timestamp attached to the GoPro video filename for automated data fusion in post.

I can't use systemd to send a shutdown command via Open GoPro's USB connection upon shutdown, because the Raspberry Pi is powered via GPIO by the robotic platform and power is always removed 'suddenly' from the system without shutdown commands being issued by the Pi.

Setting NAKE=1 and removing the battery isn't going to be viable either because the videos get pushed through an automated processing pipeline at the end of each day and I need to avoid corrupted videos at all costs (even if all videos except the last one of the day have been repaired by the GoPro on subsequent power-ons).

Any help or ideas would be appreciated.

**Comment by dnewman-gpsw** [comment, marked as answer]
Only on HERO11 or 12, although the feature is planned to be rolled back to 10 in a month or so.  You could have a boot script that runs all the time, that uses the new scripting language to monitor for recording and no USB power. Set up with: https://gopro.github.io/labs/build/ 

![image](https://github.com/gopro/labs/assets/10856084/060bd980-ec9b-4be4-a816-1664f80647f3)
![Close and Shutdown on No Power-3_26_2024, 07_17_14](https://github.com/gopro/labs/assets/10856084/02429f6c-c12c-442d-9c03-cf9bcd8a7861)

**Reply by AdamVen** [reply]
That's exciting news. A Labs firmware update enabling camera status conditionals on the Hero10 Black will make it viable for our organization to lock into the GoPro ecosystem long-term.

Your solution appears to be exactly what we need. I'm looking forward to implementing it upon release of the new firmware.

Thanks, and keep up the great work.
